### PR TITLE
core: Add choice of distro on install

### PR DESCRIPTION
**Pull request by justbur**
Ref #2926

### CODE REVIEW

**Comment by syl20bnr** [comment]
Excellent thank you :+1: 
I had to tweak some stuff to make it pleasant when first installing Spacemacs. For instance I made the default distro to be `spacemacs-core` so the wizard is triggered sooner. I also move `ido-vertical-mode` to `spacemacs-core` in order to keep the wizard questions easier to read and select (<kbd>C-j</kbd>, <kbd>C-k</kbd> enabled etc...). I added the distro name in the injected version on the home buffer.

Cherry-picked into develop branch, you can safely delete your branch.

**Comment by justbur** [comment]
Partially reverted in 098e43c8ff56ef146ce4fc10633522a3447c8ab4 

I think we need to figure out a way of doing this without making the default distro spacemacs-core, which relies on people setting their distro explicitly (and reading the options carefully)

**Comment by syl20bnr** [comment]
@justbur @TheBB Ideally we should ask for the distro right after the bootstrap, I tried it and there is no technical limitation to do this but there is no vertical ido at this time of loading so the choices are really hard to read. This is why I moved vertical ido to spacemacs-core (which I think should stay there). The change for default spacemacs-core was made to prevent a silly case where spacemacs installs the spacemacs distro and the user chooses spacemacs-core which ends up deleting 63 packages which have been installed for nothing.

So in a nutshell we should install the dotfile at the beginning of init.el, the only issue to solve is to have a readable ido.

**Comment by justbur** [comment]
@syl20bnr I think that's probably the best solution, but I wanted to say that I see `spacemacs-core` as being for people who understand spacemacs and what they want from it. For those people I don't think it's too much to ask to have them create the dotfile themselves without the wizard. That doesn't mean that's the "right" approach though.

**Comment by syl20bnr** [comment]
@justbur I moved the wizard before the layers syncing, we'll resolve the crappy horizontal ido later.
